var/spack/repos/builtin/packages/spatialdata/package.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Spatialdata(AutotoolsPackage):
+    """Spatialdata is a C++ library for
+
+    interpolating values for spatially distributed data,
+    converting coordinates among geographic projections using Proj,
+    nondimensionalization of quantities,
+    specification of units via Pyre (optional).
+
+    This library is used in the finite-element code PyLith
+    (https://github.com/geodynamics/pylith). The primary focus is specification
+    of parameters that vary in space, such as values for boundary conditions
+    and parameters of constitutive models. This provides a specification of
+    these parameters independent of the discretization."""
+
+    homepage = "https://geodynamics.org/resources/spatialdata/"
+    url = "https://github.com/geodynamics/spatialdata/archive/refs/tags/v3.1.0.tar.gz"
+
+    license("MIT", checked_by="downloadico")
+
+    version(
+        "develop",
+        git="https://github.com/geodynamics/spatialdata/",
```

Comment:
Can you move the `git` attribute to be class scoped?

```
url = "..."
git = "..."

license(...)

version("develop", branch="develop", submodules="true")
```

That way users will be able to use `spack isntall spacialdata@git.REF` to install from other refs in the git repo. It's also generally best practice to have the `preferred` version not be a git branch, is there a reason we should do otherwise specifically for this package?

Reply:
Thanks!  Done.